### PR TITLE
Put values into beng-clip-cnt sensor

### PR DESCRIPTION
**Pull request by bmerry**
Checklist (if not applicable, edit to add `(N/A)` and mark as done):

- [x] (n/a) If dependencies are added/removed: update `setup.cfg` and `.pre-commit-config.yaml`
- [x] (n/a) If modules are added/removed: use `sphinx-apidoc -efo doc/ src/` to update files in `doc/`
- [x] Ensure copyright notices are present and up-to-date
- [x] (n/a) If qualification tests are changed: attach a sample qualification report
- [x] (n/a) If design has changed: ensure documentation is up to date
- [x] (n/a) If ICD-defined sensors have been added: update `fake_servers.py` in katsdpcontroller to match

Closes NGC-1173. Note that the sensor has already been wired up to katsdpcontroller, so no changes are needed there.

### CODE REVIEW

File: src/katgpucbf/xbgpu/bsend.py [hunk 1]
```diff
@@ -194,30 +197,41 @@ def _inc_counters(
...
         output_names
-            List of beam stream names that are enabled for transmission for
-            this `future`.
+            List of beam stream names
         saturated
             Saturation count for the chunk for each stream in `output_names`.
+        sensors
+            Server's katcp sensors
+        sensor_timestamp
+            Timestamp (UNIX time) to use for sensor update
```
Comment (amishatishpatel) [inline src/katgpucbf/xbgpu/bsend.py:241]: Full stops at the end of these descriptions, please (at least to fit with the rest of them).

File: src/katgpucbf/xbgpu/bsend.py [hunk 2]
```diff
@@ -194,30 +197,41 @@ def _inc_counters(
...
+            # Don't update output counters if we didn't successfully transmit the data.
+            n_frames_sent = 0
+        # int casts are because np.prod returns np.int64 which is
+        # incompatible with the type annotations for Prometheus.
+        # Multiply across dimensions to get total bytes
+        byte_count = int(np.prod(data_shape)) * data_dtype.itemsize * n_frames_sent
+        # Multiply across the first two dimensions to get complex sample count
+        sample_count = int(np.prod(data_shape[:-1])) * n_frames_sent
+        for i, output_name in enumerate(output_names):
+            clipped = int(saturated[i])  # Convert from numpy type to plain Python type
```
Comment (amishatishpatel) [inline src/katgpucbf/xbgpu/bsend.py:255]: Really bikeshedding now. Do you think we can consolidate this conversing/casting comment with the old ones above? Line 220+ kinda discuss the same thing.

File: test/xbgpu/test_engine.py [hunk 3]
```diff
@@ -470,6 +470,12 @@ def prom_get(name: str) -> float:
         assert prom_get("output_b_samples_total") == n_beam_heaps_sent * heap_samples
         assert saturated_low[i] <= prom_get("output_b_clipped_samples_total") <= saturated_high[i]
 
+        # Check that sensor updates match Prometheus updates
```
Comment (amishatishpatel) [inline test/xbgpu/test_engine.py:473]: Not to be a pain, but you're not checking updates plural here, are you? You're checking the last update in the list of updates.
* Nothing wrong with that, it clearly makes sense.
* I'd request you update/tweak the comment to reflect that, please.

File: test/xbgpu/test_engine.py [hunk 4]
```diff
@@ -1150,14 +1156,10 @@ def heap_factory(batch_index: int) -> list[spead2.send.HeapReference]:
...
-            == n_channels_per_substream * n_baselines
-        )
-        assert (
-            prom_diff.get_sample_diff("output_x_clipped_visibilities_total", {"stream": "bcp1"})
-            == n_channels_per_substream * n_baselines
-        )
+        n_vis = n_channels_per_substream * n_baselines
+        assert prom_diff.get_sample_diff("output_x_visibilities_total", {"stream": "bcp1"}) == n_vis
+        assert prom_diff.get_sample_diff("output_x_clipped_visibilities_total", {"stream": "bcp1"}) == n_vis
+        assert xbengine.sensors["bcp1.xeng-clip-cnt"].value == n_vis
```
Comment (amishatishpatel) [inline test/xbgpu/test_engine.py:1162]: Ha, nice one. I do feel bad for asking you to loop this over each `corrprod_output`. Not a priority, I'll create an `if-bored` ticket.

**Review by amishatishpatel — CHANGES_REQUESTED** [review]
Couple requests for tweaks. Otherwise, fantastic.

**Comment by amishatishpatel** [comment]
Apologies, no idea how some comments came through before others. Maybe I submitted them as Single Comments rather than part of the review.

File: src/katgpucbf/xbgpu/bsend.py (hunk 2 above)
Comment (bmerry) [reply src/katgpucbf/xbgpu/bsend.py:255]: I'll just delete the comment here since it's redundant.

File: test/xbgpu/test_engine.py (hunk 4 above)
Comment (bmerry) [reply test/xbgpu/test_engine.py:1162]: I'm just being consistent with what's already been done for Prometheus.

**Review by amishatishpatel — APPROVED** [review]

**Pull request merged**